src/cli/cli-config.ts

```diff
@@ -142,6 +149,15 @@ export class CmdCliConfig {
    */
   static async detectOrDownload(cmd: string): Promise<string> {
     try {
+      if (!macUser.get('m1MacUser') && !macUser.get('notM1MacUser') && Platform.OS === 'darwin') {
```

Comment:
I don't have a mac so i cannot test this thing at all but by looking at the code it's seems that M1 is arm64. Why don't you just do `if (process.arch === 'arm64' and darwin) download the cli arm64 `?